models/src/Bombs.cpp

```diff
@@ -23,16 +23,51 @@ void Bombs::placeBomb(const Player &player, Map &map)
 	}
 }
 
+const float BOMB_RANGE = 1; // TODO: change based on player powerups
 void Bombs::update(float deltaTime, Map &map)
 {
 	for (sBomb &bomb : this->_bombs)
 	{
 		bomb.timeLeft -= deltaTime;
+        // If current bomb is on a flame, detonate the bomb
+        if (map.tileAt(bomb.position) == Tile::Flame)
+        {
+            bomb.timeLeft = -1;
+        }
 		if (bomb.timeLeft < 0)
 		{
-			// Explode logic here
-			map.setTile(bomb.position, Tile::Clear);
+            for (int i = 1; i < BOMB_RANGE + 1; ++i)
```

Comment:
I really like this solution, the only issue is that you are piercing. Please add some logic to prevent the propagation through solids